src/cpuid.cpp

```diff
@@ -46,7 +46,14 @@ POSSIBILITY OF SUCH DAMAGE.
 #include <cstring> // for std::memset
 #endif
 
-#if TORRENT_HAS_ARM && defined __linux__ && (TORRENT_HAS_ARM_NEON || TORRENT_HAS_ARM_CRC32)
+#if defined __GLIBC__ && (__GLIBC__ > 2 || (__GLIBC__ == 2 && __GLIBC_MINOR__ >= 16))
```

Comment:
do you mind adding here `|| TORRENT_ANDROID`?

my understanding is that it was added in `glibc` but it's _not_ `glibc` specific, in fact, that's the documented way in android to get the CPU features.